src/main/java/com/thebois/models/world/World.java

```diff
@@ -20,39 +24,42 @@
 /**
  * World creates a matrix and keeps track of all the structures and resources in the game world.
  */
-public class World implements IWorld, IFinder {
+public class World implements IWorld, IFinder, IResourceFinder {
 
     private final ITerrain[][] terrainMatrix;
     private final IStructure[][] structureMatrix;
     private final IResource[][] resourceMatrix;
     private final ITile[][] canonicalMatrix;
     private final int worldSize;
+    private final Random random;
 
     /**
      * Initiates the world with the given size.
      *
      * @param worldSize The amount of tiles in length for X and Y, e.g. worldSize x worldSize.
      * @param seed      The seed used to generate the world.
+     * @param random    A generator of random numbers.
      */
-    public World(final int worldSize, final int seed) {
+    public World(final int worldSize, final int seed, final Random random) {
         this.worldSize = worldSize;
+        this.random = random;
```

Comment:
Will we use a different `Random`? Because the only thing we do is create it in pawntastic and immediately send it to world